core/base/src/main/java/org/eclipse/dataspaceconnector/core/base/BoundedCommandQueue.java

```diff
@@ -22,49 +23,45 @@
 import java.util.concurrent.ArrayBlockingQueue;
 import java.util.concurrent.BlockingQueue;
 
-/**
- * In-memory implementation of the {@link CommandQueue} that is backed by a bounded queue. This means that attempting
- * to add elements to an already full queue will fail and raise an {@link IllegalStateException}.
- * <p>
- * This queue is threadsafe, so every operation on it can be assumed atomic.
- */
-public class BoundedCommandQueue implements CommandQueue {
-    private final BlockingQueue<Command> queue;
-
+public class BoundedCommandQueue<C extends Command> implements CommandQueue<C> {
+    
+    private final BlockingQueue<C> queue;
+    
     public BoundedCommandQueue(int bound) {
         queue = new ArrayBlockingQueue<>(bound);
     }
-
+    
     @Override
-    public void enqueue(Command element) {
+    public void enqueue(C element) {
         //add will throw an IllegalStateException if the queue exceeds its capacity
         queue.add(element);
     }
-
+    
     @Nullable
     @Override
     public Command dequeue() {
         return queue.poll();
```

Comment:
I'd recommend using [BlockingQueue#poll(time, unit)](https://docs.oracle.com/javase/7/docs/api/java/util/concurrent/BlockingQueue.html#poll(long,%20java.util.concurrent.TimeUnit)) which gives us blocking functionality out of the box. Duration and TimeUnit could be set via the BoundedCommandQueue constructor

<sub> Denis Neuling <denis.neuling@daimler.com>, Daimler TSS GmbH, [legal info/Impressum](https://github.com/Daimler/daimler-foss/blob/master/LEGAL_IMPRINT.md)

Reply:
Actually I think this method can be removed since it's not used by anyone, corrently the `dequeue(amount)` is used